packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-load-balancer.ts

```diff
@@ -208,10 +209,7 @@ export class NetworkLoadBalancer extends BaseLoadBalancer implements INetworkLoa
    * @deprecated use ``NetworkTargetGroup.metricHealthyHostCount`` instead
    */
   public metricHealthyHostCount(props?: cloudwatch.MetricOptions) {
-    return this.metric('HealthyHostCount', {
-      statistic: 'Average',
-      ...props,
-    });
+    return this.cannedMetric(NetworkELBMetrics.healthyHostCountAverage, props);
```

Comment:
Oooh!

So this is actually another case of the dimensionality being wrong (like the previous DynamoDB UserErrors example), which is why these methods are deprecated. The (un)healthyHostCount metrics require the `TargetGroup` dimension. The correct methods are in the `NetworkTargetGroup` class. I'm guessing the canned metrics source file has this wrong too?

It doesn't harm anything to use the canned metrics here, I suppose, but it'd be nice to have the metric fixed and use the canned metrics in the NetworkTargetGroup methods instead.

Reply:
I know. I've reported the error to CloudWatch Explorer team.

That's the downside of this: we now have an extra iteration to do for fixes.

Might as well not have changed this metric function I suppose.